docs/pages/world/modules/keysintable.mdx

```diff
@@ -0,0 +1,239 @@
+import { Callout } from "nextra/components";
+
+import { CollapseCode } from "../../../components/CollapseCode";
+
+# Keys in Table Module
+
+[The `KeysInTable` module](https://github.com/latticexyz/mud/tree/main/packages/world-modules/src/modules/keysintable) tracks which keys are used with a specific table.
+This is useful if you need to be able to iterate through all the records of a table onchain.
+
+<Callout type="warning" emoji="💰">
+It costs additional gas for every write to the table affected.
+If you just need this functionality offchain it is best to look at the data offchain instead.
+
+Also, this is a _generic_ solution.
+You can usually save gas by implementing a solution that is specific to your use case, using [hooks](/store/store-hooks).
+
+</Callout>
+
+## Deployment
+
+It can be deployed multiple times, each time for a different table.
+For example, here is a `mud.config.ts` that deploys it for the `Tasks` table in [the React template](https://github.com/latticexyz/mud/tree/main/templates/react).
+
+<CollapseCode>
+
+```typescript filename="mud.config.ts" copy showLineNumbers {2,17-32}
+import { defineWorld } from "@latticexyz/world";
+import { resourceToHex } from "@latticexyz/common";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Tasks: {
+      schema: {
+        id: "bytes32",
+        createdAt: "uint256",
+        completedAt: "uint256",
+        description: "string",
+      },
+      key: ["id"],
+    },
+  },
+  modules: [
+    {
+      root: true,
+      artifactPath: "@latticexyz/world-modules/out/KeysInTableModule.sol/KeysInTableModule.json",
+      args: [
+        {
+          type: "bytes32",
+          value: resourceToHex({
+            type: "table",
+            namespace: "app",
+            name: "Tasks",
+          }), // end of value
+        }, // end of arg
+      ], // end of args
+    }, // end of module
+  ], // end of modules
```

Comment:
these comments are unnecessary noise, can we remove?

if we need to clarify this, we can:
- lift these out into functions/variables
- add syntax highlighting/hovers for brackets/nesting (like VSCode)

Reply:
> add syntax highlighting/hovers for brackets/nesting (like VSCode)

Do you know how to do that? That will be a good addition.